File: backend/src/main/java/com/knowvault/summary/Summary.java
package com.knowvault.summary;

import com.knowvault.content.Content;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "summaries")
@Getter @Setter @NoArgsConstructor
public class Summary {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "content_id")
    private Content content;

    @Column(name = "summary_type")
    private String summaryType;

    @Column(columnDefinition = "TEXT")
    private String summary;

    @Column(name = "key_points", columnDefinition = "TEXT")
    private String keyPoints;

    @Column(name = "generated_at")
    private String generatedAt;
}

File: frontend/src/types/summary.ts
export type SummaryType = 'short' | 'medium' | 'detailed';

export interface Summary {
  id: number;
  contentId: number;
  summaryType: SummaryType;
  summary: string;
  keyPoints: string[];
  generatedAt: string;
}

export interface SummaryGenerateRequest {
  type: SummaryType;
}
